### ISSUE TITLE
copy dependency management for bootstrap modules to quarkus-bom

### ISSUE THREAD

**Issue by gemmellr** [quarkusio/quarkus#12628]
The quarkus-universe-bom can contain dependency management entries coming from extensions which override those used by the core quarkus bootstrap modules, as the core quarkus-bom may not contain dependency management entries for them while the platform quarkus-universe-bom could due to an extension doing so.

While looking into and discussing [1] an extension build failure I was seeing while trying the build out using Quarkus 1.9.0.CR1 core, due to an enforcer dependency convergence check failing for plexus-classworlds (which turned out to be a non-issue in the end, resolved by updating the Maven install on the little-used system from 3.6.0 to 3.6.3), I had also noticed that while the bootstrap modules have dependencyManagement meaning plexus-classworlds 2.6.0 should be used, the core quarkus-bom does not mention it, whilst the platform quarkus-universe-bom does but with the older 2.5.2 version.

During the discussion @aloubyansky suggested that the bootstrap dependency management should be copied to quarkus-bom and asked I raise an issue for it.

[1] https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/plexus-classworlds.20dep.20convergence.20with.201.2E9.2E0.2ECR1

**Comment by aloubyansky** [comment]
While it does make sense to go over the bootstrap dependencies and add the key ones to quarkus-bom, the classworlds dependency is an interesting case in that, the bootstrap doesn't actually use its classes at runtime except for satisfying the classloading requirements, i.e. if a class from the classworlds isn't on the classpath, the class that's actually used will fail to load. So, in this case, requiring a specific version of the classworlds will only make sense to satisfy the enforcer rules.
Otherwise, perhaps, we can configure the bootstrap dependencies in a manner to not require a direct dependency the classworlds.

**Comment by gemmellr** [comment]
Yep, as mentioned in the discussion the version didn't really make any difference to the build in the end, it worked fine once either the convergence check was disabled or I updated the env to Maven 3.6.3.

The only mentions of it in quarkus-bom are in some of the other deps excluding plexus-classworlds, its just that quarkus-bootstrap-parent does dep manage a specific 2.6.0 version for it, while quarkus-universe-bom then dep manages (via some extension) a different older 2.5.2 version. 

For whatever reason, when using just quarkus-bom in my extension build, Maven 3.6.0 falls over when dealing with the version from quarkus-bootstrap-parent, and the different version (but also 2.5.2) from another of the bootstrap deps. On trying it out now, the same convergence issue did already present itself when building the extension against Quarkus 1.8.x and 1.7.x and potentially earlier, I obviously just havent built the extension with those versions / on that system in a while (as I dont normally use it for dev), and hadn't noticed the Maven version was a little behind and causing the issue, whilst the build has worked fine on my regular dev system as it already had Maven 3.6.3 on it. Noticing the universe bom defines a different classworlds version was just something I spotted while looking around while hunting for an explanation.
